lib/api/apiUtils/object/versioning.js

```diff
@@ -434,6 +434,43 @@ function preprocessingVersioningDelete(bucketName, bucketMD, objectMD, reqVersio
     return options;
 }
 
+/**
+ * Keep metadatas when the object is restored from cold storage
+ * but remove the specific ones we don't want to keep
+ * @param {object} objMD - obj metadata
+ * @param {object} metadataStoreParams - custom built object containing resource details.
+ * @return {undefined}
+ */
+function restoreMetadatas(objMD, metadataStoreParams) {
+    /* eslint-disable no-param-reassign */
+    const userMDNotToKeep = ['x-amz-meta-scal-s3-restore-attempt'];
+    // We need to keep user metadata and tags
+    Object.keys(objMD).forEach(key => {
+        if (key.startsWith('x-amz-meta-')  && !userMDNotToKeep.includes(key)) {
+            metadataStoreParams.metaHeaders[key] = objMD[key];
+        }
+        if (key.startsWith('content-') && !userMDNotToKeep.includes(key)) {
+            metadataStoreParams[key] = objMD[key];
+        }
+    });
+
+    if (objMD.replicationInfo?.status === 'COMPLETED') {
+        metadataStoreParams.replicationInfo = objMD.replicationInfo;
+    }
+
+    if (objMD.legalHold) {
+        metadataStoreParams.legalHold = objMD.legalHold;
+    }
+
+    if (objMD.acl) {
+        metadataStoreParams.oldAcl = objMD.acl;
```

Comment:
why `oldAcl`, and not just `acl` like the other?
(i don't see an already used `acl` field in here, but maybe i missed it)

Reply:
acl is a global variable in service.js because we require it
`const acl = require('./metadata/acl');`

I wouldn't be able to create a variable like this if the field was named acl
```
const { objectKey, authInfo, size, contentMD5, metaHeaders,
            contentType, cacheControl, contentDisposition, contentEncoding,
            expires, multipart, headers, overrideMetadata, log,
            lastModifiedDate, versioning, versionId, uploadId,
            tagging, taggingCopy, replicationInfo, defaultRetention,
            dataStoreName, creationTime, retentionMode, retentionDate,
            legalHold, originOp, updateMicroVersionId, archive, oldReplayId,
            deleteNullKey, oldAcl } = params;
```

That said, I can still name it acl and use it as `params.acl`

What do you prefer?

Reply:
personally, I would prefer to keep the name of the field clean (and possibly using another name when importing the package, to avoid problems)

not sure what others think though, if anyone knows "good practice" for this case, please share!